Assets/Global/Scripts/RoomLogic/RoomTransitionDoor.cs

```diff
@@ -16,18 +16,27 @@ public class RoomTransitionDoor : Interactable
     [SerializeField] private Animator animator;
     [SerializeField] private MeshRenderer doorMesh;
     [SerializeField] private string nextRoomName;
-    private bool isTransitioning = false;
+    public int nextRoomid;
+
     public CrossfadeController crossfadeController;
     public PlayerSpawnPoint playerSpawnPoint;
+    public DoorManager doorManager;
     
     private string currentScenename;
  
 
     private void Awake()
     {
+        IsDisabled = IsDisabled; // ugly fix so maybe we have to change in the future
         GlobalReference.SubscribeTo(Events.ROOM_FINISHED, RoomFinished);
```

Comment:
do you also unsubscribe to this event somewhere? if not memory will be leaked maybe if you don't unsubscribe when destroyed.

Reply:
This is true.
I did not think of this yet, but you are right. In the long run this will have inpact.